ktor-network/jvmAndNix/src/io/ktor/network/sockets/SocketAddress.kt

```diff
@@ -4,8 +4,31 @@
 
 package io.ktor.network.sockets
 
-public sealed class SocketAddress
+public expect sealed class SocketAddress
```

Comment:
just for information: this change isn't source compatible for code written in `common` source set where exhaustive when is used

Reply:
How exactly is this incompatible? The implementations of the sealed classes are also declared in the same file.

Reply:
still, if `sealed class` is `expect` - it will be not exhausted in common code, as platforms can declare additional implementations, but common code doesn't know about them. 
So this code in common sourceset will not compile now after this changes:
```
val address: SocketAddress  = TODO()
val something = when(address) {
  is InetSocketAddress -> 1
  is UnixSocketAddress -> 2
}
```

Reply:
https://kotlinlang.org/docs/sealed-classes.html#sealed-classes-and-when-expression
specifically info in green box :)

Reply:
it can be used in kotlin code base in platform specific code, like jvm, or nix - no error will be there, the only issue is with common code

Reply:
Thanks for the links, I see. Initially I tried to declare only InetSocketAddress as an expect/actual but that fails with the screenshot above. Do you happen to have any suggestions how to get this working retaining compatibility?

Reply:
as far as I know, it's not possible to do it without breaking of source or binary compatibility...
now I think it depends on ktor team, deciding if it's acceptable to not allow to do exhaustive when on this class in common code or not

Reply:
Yes I will await their response I guess. In my opinion a minor source breaking change is less important than the UDP issue that this solves on Windows. Anyway, thanks for the review, appreciate it.

Reply:
Im always happy to help :)

Reply:
For me, it looks like an MPP issue: all implementations are available at compile time. 

I also think it's ok to make this change. @rsinukov what do you think?

Reply:
I agree.

Reply:
Okay thanks, keeping it as is.

Reply:
> still, if `sealed class` is `expect` - when will be not exhausted in common code, as platforms can declare additional implementations, but common code doesn't know about them. So this code in common sourceset will not compile after this changes:
> 
> ```
> val address: SocketAddress  = TODO()
> val something = when(address) {
>   is InetSocketAddress -> 1
>   is UnixSocketAddress -> 2
> }
> ```

Just got bitten by that while bumping to Ktor 2.0.3... :D